src/components/FormContext.tsx

```diff
@@ -31,6 +33,10 @@ export const FormContext: FC<FormContextProps> = props => {
     submitFocusError: submitFocusError,
   })
 
+  if (typeof onChange === 'function') {
+    onChange(methods.watch())
```

Comment:
This one returns a flat object, I think we should return the object as expected by the user: 
```suggestion
    onChange(getObjectFromForm(props.schema, methods.watch()))
```